ansible/TestbedProcessing.py

```diff
@@ -484,47 +477,59 @@ def makeLab(data, devices, testbed, outfile):
-
-                        try: #get voq_inband_ipv6
-                            voq_inband_ipv6 = dev.get("voq_inband_ipv6")
-                            if voq_inband_ipv6 is not None:
-                               entry += "\tvoq_inband_ipv6=" + str( voq_inband_ipv6 )
-                        except AttributeError:
-                            print("\t\t" + host + " voq_inband_ipv6 not found")
-
-                        try: #get voq_inband_intf
-                            voq_inband_intf = dev.get("voq_inband_intf")
-                            if voq_inband_intf is not None:
-                               entry += "\tvoq_inband_intf=" + str( voq_inband_intf )
-                        except AttributeError:
-                            print("\t\t" + host + " voq_inband_intf not found")
-
-                        try: #get voq_inband_type
-                            voq_inband_type = dev.get("voq_inband_type")
-                            if voq_inband_type is not None:
-                               entry += "\tvoq_inband_type=" + str( voq_inband_type )
-                        except AttributeError:
-                            print("\t\t" + host + " voq_inband_type not found")
-
                         try: #get switch_type
                             switch_type = dev.get("switch_type")
                             if switch_type is not None:
-                               entry += "\tswitch_type=" + str( switch_type )
-                        except AttributeError:
-                            print("\t\t" + host + " switch_type not found")
+                                entry += "\tswitch_type=" + str( switch_type )
+                                if switch_type == 'voq' and card_type != 'supervisor':
+                                    if num_asics is None:
+                                        num_asics = 1
+
+                                    # All fields are a list. For single asic the list is of size 1.
+                                    switchids = dev.get("switchids")                # switchids
+                                    voq_inband_ip = dev.get("voq_inband_ip")        # voq_inband_ip
+                                    voq_inband_ipv6 = dev.get("voq_inband_ipv6")    # voq_inband_ipv6
+                                    voq_inband_intf = dev.get("voq_inband_intf")    # voq_inband_intf
+                                    voq_inband_type = dev.get("voq_inband_type")    # voq_inband_type
+                                    max_cores = dev.get("max_cores")                # max cores
+                                    if num_asics > 1:
+                                        lo4096_ip = dev.get("loopback4096_ip")      # loopback4096_ip
+                                        lo4096_ipv6 = dev.get("loopback4096_ipv6")  # loopback4096_ipv6
+
+                                    # Set defaults
+                                    if switchids is None:
```

Comment:
move these to line 517 to be closer to 
start_switchid += num_asics

Reply:
Addressed in commit 7daf2e0